cli/search/search.go

```diff
@@ -37,178 +32,24 @@ type SearchCtx struct {
-			continue
-		}
-		versions = append(versions, version)
-	}
-
-	sort.Stable(version.VersionSlice(versions))
-
-	return versions, nil
 }
 
-// GetCommitFromGitLocal returns hash or pr/ID info from specified local git repo.
-func GetCommitFromGitLocal(repo string, input string) (string, error) {
-	if _, err := exec.LookPath("git"); err != nil {
-		return "", fmt.Errorf("unable to get commits: `git` command is missing")
-	}
-
-	isPullRequest, pullRequestID := util.IsPullRequest(input)
-
-	if isPullRequest {
-		commandStr := "pull/" + pullRequestID +
-			"/head:" + input
-		cmd := exec.Command("git", "fetch", "origin", commandStr)
-		cmd.Dir = repo
-		err := cmd.Run()
-		if err != nil {
-			return "", err
-		}
-	}
-
-	cmd := exec.Command("git", "show", input, "--quiet")
-	cmd.Dir = repo
-
-	output, err := cmd.Output()
-	if err != nil {
-		return "", err
-	}
-
-	lines := strings.Split(strings.TrimSpace(string(output)), "\n")
-
-	hash := strings.Split(lines[0], " ")[1]
-
-	return hash, nil
-}
-
-// GetCommitFromGitRemote returns hash or pr/ID info from specified remote git repo.
-func GetCommitFromGitRemote(repo string, input string) (string, error) {
-	if _, err := exec.LookPath("git"); err != nil {
-		return "", fmt.Errorf("unable to get commits: `git` command is missing")
-	}
-
-	tempRepoPath, err := os.MkdirTemp("", "tt_install_repo")
-	if err != nil {
-		return "", fmt.Errorf("failed to get commits from %q: %s", repo, err)
-	}
-
-	defer os.RemoveAll(tempRepoPath)
-
-	cmd := exec.Command("git", "clone", "--filter=blob:none", "--no-checkout",
-		"--single-branch", repo, tempRepoPath)
-
-	err = cmd.Run()
-	if err != nil {
-		return "", fmt.Errorf("unable to get commits: git clone failed: %w", err)
-	}
-
-	return GetCommitFromGitLocal(tempRepoPath, input)
-}
-
-// GetVersionsFromGitLocal returns sorted versions list from specified local git repo.
-func GetVersionsFromGitLocal(repo string) ([]version.Version, error) {
-	versions := []version.Version{}
-
-	if _, err := exec.LookPath("git"); err != nil {
-		return nil, fmt.Errorf("'git' is required for 'tt search' to work")
-	}
-
-	output, err := exec.Command("git", "-C", repo, "tag").Output()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get versions from %s: %s", repo, err)
-	}
-
-	lines := strings.Split(strings.TrimSpace(string(output)), "\n")
-
-	// No tags found.
-	if len(lines) == 1 && lines[0] == "" {
-		return versions, nil
-	}
-
-	for _, line := range lines {
-		version, err := version.Parse(line)
-		if err != nil {
-			continue
-		}
-		if isMasked(version) && strings.Contains(repo, "tarantool") {
-			continue
-		}
-		versions = append(versions, version)
-	}
-
-	sort.Stable(version.VersionSlice(versions))
-
-	return versions, nil
+// NewSearchCtxWithMock creates a new SearchCtx with a mock handlers for testing.
+func NewSearchCtxWithMock(informer platformInformer, doer apiDoer) SearchCtx {
+	sCtx := NewSearchCtx()
+	sCtx.platformInformer = informer
+	sCtx.apiDoer = doer
+	return sCtx
```

Comment:
To be honest, I am against the approach. It should be possible to test a public API through this public API without additional methods. Otherwise, there is no point in all this stuff and you can test a private API.

It is better to change the current public API so that it is oriented to dependency injection and you will not require the test-helper method.